src/utlis/listener.ts

```diff
@@ -0,0 +1,80 @@
+import { storeToRefs } from "pinia";
+import { Event, UnlistenFn } from "@tauri-apps/api/event";
+import { ref } from "vue";
+import { ManagedSerialPortsEvent } from "@/events";
+import { PacketData } from "@/models/intern/packet-data";
+import {
+  listenPacketEvent,
+  listenSerialPortEvent,
+  listenThemeChangedEvent,
+} from "@/api/listener";
+import { useAppStore } from "@/stores/app";
+import { useTheme } from "vuetify";
+
+export const useListener = (app = useAppStore()) => {
+  const { managedSerialPorts } = storeToRefs(app);
+  const { addPacket, getSerialPorts } = app;
+  const theme = useTheme();
+
+  const themeChangedEventListener = ref<UnlistenFn>();
+  const serialPortEventListener = ref<UnlistenFn>();
+  const serialPortPacketEventListener = ref<UnlistenFn>();
+
+  const onThemeEventListenerTrigger = (event: Event<string>) => {
+    const themeName = event.payload;
+    if (themeName === "dark" || themeName === "light") {
+      theme.global.name.value = themeName;
+    }
+  };
+
+  const onSerialPortPacketEventListener = (event: Event<any>) => {
+    const packet = event.payload.packet;
+
+    const packetData: PacketData = {
+      packetDirection: packet.packetDirection,
+      timestampMillis: packet.timestampMillis,
+    };
+
+    addPacket(packet.portName, packetData);
+  };
+
+  const onSerialPortEventListener = (event: Event<ManagedSerialPortsEvent>) => {
+    managedSerialPorts.value = event.payload.ports;
+  };
+
+  const setupListeners = async () => {
+    themeChangedEventListener.value = await listenThemeChangedEvent(
+      onThemeEventListenerTrigger
+    );
+
+    serialPortEventListener.value = await listenSerialPortEvent(
+      onSerialPortEventListener
+    );
+
+    serialPortPacketEventListener.value = await listenPacketEvent(
+      onSerialPortPacketEventListener
+    );
+
+    getSerialPorts();
+  };
+
+  const cleanupListeners = () => {
+    if (themeChangedEventListener.value) {
+      themeChangedEventListener.value();
+      themeChangedEventListener.value = undefined;
+    }
+    if (serialPortEventListener.value) {
+      serialPortEventListener.value();
+      serialPortEventListener.value = undefined;
+    }
+    if (serialPortPacketEventListener.value) {
+      serialPortPacketEventListener.value();
+      serialPortPacketEventListener.value = undefined;
```

Comment:
Werden Listener geschlossen, wenn man value auf undefined setzt?

Reply:
Ja, die Event Listener werden in der Zeile darüber _Revertet_